File: Java - Microservices with Spring Boot, Spring Cloud/Glide-Smart-Bike-Rental-System-main/glide-frontend/src/services/bikeService.js
import api from "./api";

// Get all bikes
export const getAllBikes = async () => {

    const response = await api.get("/api/bikes");

    return response.data;

};

// Get bike by ID
export const getBikeById = async (id) => {

    const response = await api.get(
        `/api/bikes/${id}`
    );

    return response.data;

};

// Add bike
export const addBike = async (bikeData) => {

    const response = await api.post(
        "/api/bikes",
        bikeData
    );

    return response.data;

};

// Update bike
export const updateBike = async (id, bikeData) => {

    const response = await api.put(
        `/api/bikes/${id}`,
        bikeData
    );

    return response.data;

};

// Delete bike
export const deleteBike = async (id) => {

    const response = await api.delete(
        `/api/bikes/${id}`
    );

    return response.data;

};

// Get bikes by station
export const getBikesByStation = async (stationId) => {

    const response = await api.get(
        `/api/bikes/station/${stationId}`
    );

    return response.data;

};

File: Java - Microservices with Spring Boot, Spring Cloud/Glide-Smart-Bike-Rental-System-main/glide-backend/src/main/java/com/glide/service/BikeService.java
package com.glide.service;

import com.glide.dto.BikeRequest;
import com.glide.dto.BikeResponse;

import java.util.List;

public interface BikeService {

    BikeResponse addBike(BikeRequest request);

    List<BikeResponse> getAllBikes();

    BikeResponse getBikeById(Long id);

    BikeResponse updateBike(Long id, BikeRequest request);

    void deleteBike(Long id);

    List<BikeResponse> getBikesByStation(Long stationId);
}